libcore/ConnectServerDialog.vala

```diff
@@ -0,0 +1,530 @@
+
+        var password_label = new Gtk.Label (_("Password:"));
+        password_label.xalign = 1;
+        password_entry = new Gtk.Entry ();
+        password_entry.visibility = false;
+        password_entry.bind_property ("visible", password_label, "visible", GLib.BindingFlags.DEFAULT);
+
+        remember_checkbutton = new Gtk.CheckButton.with_label (_("Remember this password"));
+        password_entry.bind_property ("visible", remember_checkbutton, "visible", GLib.BindingFlags.DEFAULT);
+
+        cancel_button = new Gtk.Button.with_label (_("Cancel"));
+        cancel_button.show ();
+        cancel_button.clicked.connect (on_cancel_clicked);
+
+        connect_button = new Gtk.Button.with_label (_("Connect"));
+        connect_button.no_show_all = true;
+        connect_button.show ();
+        connect_button.clicked.connect (on_connect_clicked);
+
+        continue_button = new Gtk.Button.with_label (_("Continue"));
+        continue_button.no_show_all = true;
+        continue_button.hide ();
+        continue_button.clicked.connect (on_continue_clicked);
+
+        ((Gtk.Container)(get_action_area ())).add (cancel_button);
+        ((Gtk.Container)(get_action_area ())).add (connect_button);
+        ((Gtk.Container)(get_action_area ())).add (continue_button);
+
+        key_press_event.connect ((e) => {
+            switch (e.keyval) {
+                case Gdk.Key.Return:
+                case Gdk.Key.KP_Enter:
+                    connect_button.clicked ();
+                    return true;
+
+                default:
+                    return false;
+            }
+        });
+
+        key_release_event.connect ((e) => {
+            return true;
+        });
+
+        var grid = new Gtk.Grid ();
+        grid.margin_start = grid.margin_end = 12;
+        grid.margin_bottom = 6;
+        grid.orientation = Gtk.Orientation.VERTICAL;
+        grid.row_spacing = 6;
+        grid.column_spacing = 12;
+        grid.attach (server_header_label, 0, 0, 4, 1);
+        grid.attach (server_label, 0, 1, 1, 1);
+        grid.attach (server_entry, 1, 1, 1, 1);
+        grid.attach (port_label, 2, 1, 1, 1);
+        grid.attach (port_spinbutton, 3, 1, 1, 1);
+        grid.attach (type_label, 0, 2, 1, 1);
+        grid.attach (type_combobox, 1, 2, 3, 1);
+        grid.attach (share_label, 0, 3, 1, 1);
+        grid.attach (share_entry, 1, 3, 3, 1);
+        grid.attach (folder_label, 0, 4, 1, 1);
+        grid.attach (folder_entry, 1, 4, 3, 1);
+        grid.attach (user_header_label, 0, 5, 4, 1);
+        grid.attach (domain_label, 0, 6, 1, 1);
+        grid.attach (domain_entry, 1, 6, 3, 1);
+        grid.attach (user_label, 0, 7, 1, 1);
+        grid.attach (user_entry, 1, 7, 3, 1);
+        grid.attach (password_label, 0, 8, 1, 1);
+        grid.attach (password_entry, 1, 8, 3, 1);
+        grid.attach (remember_checkbutton, 1, 9, 3, 1);
+
+        var label_sizegroup = new Gtk.SizeGroup (Gtk.SizeGroupMode.HORIZONTAL);
```

Comment:
What's this size group used for? It seems like all of these labels are in the Grid above?

Reply:
Good point.